bootstrapping/bootstrapping.go

```diff
@@ -0,0 +1,428 @@
+	bsStatus.attributes.rawData = bsConfiguration.Attributes
+
+	// Policies
+	for _, policyFile := range bsConfiguration.PolicyFiles {
+		policyStatus := new(policyStatus)
+		policyStatus.id = policyFile.ID
+		policyStatus.revisionID = policyFile.RevisionID
+		policyStatus.name = strings.Join([]string{policyFile.ID, "-", policyFile.RevisionID}, "")
+		policyStatus.filename = strings.Join([]string{policyStatus.name, ".tgz"}, "")
+		policyStatus.tarballURL = policyFile.DownloadURL
+
+		url, err := url.Parse(policyStatus.tarballURL)
+		if err != nil {
+			// TODO should it be an error?
+			return err
+		}
+		policyStatus.queryURL = strings.Join([]string{url.Path[1:], url.RawQuery}, "?")
+
+		policyStatus.tarballPath = strings.Join([]string{directoryPath, policyStatus.filename}, "")
+		policyStatus.folderPath = strings.Join([]string{directoryPath, policyStatus.name}, "")
+
+		bsStatus.policiesStatus = append(bsStatus.policiesStatus, *policyStatus)
+	}
+	log.Debug(bsStatus)
+	return nil
+}
+
+// downloadPolicyFiles For every policy file, ensure its tarball (downloadable through their download_url) has been downloaded to the server ...
+func downloadPolicyFiles(bootstrappingSvc *blueprint.BootstrappingService, bsStatus *bootstrappingStatus) error {
+	log.Debug("downloadPolicyFiles")
+
+	for _, policyStatus := range bsStatus.policiesStatus {
+		log.Debug("Downloading: ", policyStatus.tarballURL)
+		_, status, err := bootstrappingSvc.DownloadPolicyFile(policyStatus.queryURL, policyStatus.tarballPath)
+		if err != nil {
+			return err
+		}
+		if status == 200 {
+			policyStatus.downloaded = true
+			log.Debug("Uncompressing: ", policyStatus.tarballPath)
+			if err = utils.Untar(policyStatus.tarballPath, policyStatus.folderPath); err != nil {
+				return err
+			}
+			policyStatus.uncompressed = true
+		} else {
+			// TODO should it be an error?
+			log.Error("Cannot download the policy file: ", policyStatus.filename)
+		}
+	}
+	return nil
+}
+
+// cleanObsoletePolicyFiles cleans off any tarball that is no longer needed.
+func cleanObsoletePolicyFiles(directoryPath string, bsStatus *bootstrappingStatus) error {
+	log.Debug("cleanObsoletePolicyFiles")
+
+	// builds an array of currently processable files at this looping time
+	currentlyProcessableFiles := []string{bsStatus.attributes.filename} // saved attributes file name
+	for _, policyStatus := range bsStatus.policiesStatus {
+		currentlyProcessableFiles = append(currentlyProcessableFiles, policyStatus.filename) // Downloaded tgz file names
+		currentlyProcessableFiles = append(currentlyProcessableFiles, policyStatus.name)     // Uncompressed folder names
+	}
+
+	// evaluates working folder
+	files, err := ioutil.ReadDir(directoryPath)
+	if err != nil {
+		// TODO should it be an error?
+		log.Warn("Cannot read directory: ", directoryPath, err)
+	}
+
+	// removes files not regarding to any of current policy files
+	for _, f := range files {
```

Comment:
I would actually start with the contents of the directory as deletable files `deletableFiles, err :=  ioutil.ReadDir(directoryPath)` and then remove from it the attributes file and the policyfiles files